evm/src/cpu/kernel/interpreter.rs

```diff
@@ -103,6 +128,16 @@ pub(crate) fn run<'a>(
     Ok(interpreter)
 }
 
+/// Different types of Memory operations in the interpreter, and the data required to revert them.
+enum InterpreterMemOpKind {
+    /// We need to provide the context.
+    Push(usize),
+    /// If we pop a certain value, we need to push it back to the correct context when reverting.
+    Pop(U256, usize),
+    /// If we write a value at a certain address, we need to write the old value back when reverting
```

Comment:
sorry, my OCD is kicking in again 😅 (for consistency)
```suggestion
    /// If we write a value at a certain address, we need to write the old value back when reverting.
```